src/main/java/com/sixt/service/framework/rpc/backoff/RetryBackOffFunction.java

```diff
@@ -0,0 +1,35 @@
+package com.sixt.service.framework.rpc.backoff;
+
+import java.time.Duration;
+import java.util.Date;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@FunctionalInterface
+public interface RetryBackOffFunction {
+
+    Duration timeout(int retryCounter);
+
+    default void execute(int retryCounter) {
+        Duration timeout = timeout(retryCounter);
+        if (timeout == null || timeout.isNegative()) {
+            throw new IllegalArgumentException("Retry timeout cant be null or negative.");
+        }
+
+        AtomicBoolean notOverYet = new AtomicBoolean(true);
+        Long startedAt = new Date().getTime();
+
+        synchronized (notOverYet) {
```

Comment:
the primary reason I asked you to refactor this feature was this synchronized block.  1. sleeping 1ms at a time doesn't make sense.  2. why is there synchronization involved in any case?